### ISSUE TITLE
USER ISSUE: Submiting District Map at the Zoning office

### ISSUE THREAD

**Issue by WickedFreaK** [StrangeLoopGames/EcoIssues#22870]
\
**User Id: (Not logged user)**
**Version:** 0.9.4.4 beta release-273

**Steps to Reproduce:**
I was creating the law for the public roads marking them on the map. When I clicked submit, the server crashed throwing the exception.

**Expected behavior:**
save my road draft and all that

**Actual behavior:**
The following error message popped up:
```
Server encountered an exception:

<size=60.00%>Exception: InvalidOperationException
Message:Invalid Query - Query never existed or was already processed
Source:Eco.Shared

System.InvalidOperationException: Invalid Query - Query never existed or was already processed
   at Eco.Shared.Networking.RPCManager.HandleQueryResponse(BSONObject bson)
   at Eco.Plugins.Networking.NetworkServer.Eco.Shared.Networking.INetworkEventHandler.ReceiveEvent(INetClient client, NetworkEvent netEvent, BSONObject bsonObj)
   at Eco.Plugins.Networking.Client.<.ctor>b__71_4()</size>
```

**Do you have mods installed? Does the issue happen when no mods are installed?:**
no mods

**Please provide log files. Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file

**Comment by WickedFreaK** [comment]
Sorry cfor the horrible description, i hope it still helps

**Comment by WickedFreaK** [comment]
People said the server didnt crahs. it was just me who was kicked to the game menu with the exception, which said that apparewntly the server crashed

**Comment by tomesc99** [comment]
Hey @WickedFreaK ,

Thank you very much for your feedback. Did you try again after the crash? Were you able to reproduce the issue again?

If you are able to reproduce the issue again, could you provide the log files?

**Comment by WickedFreaK** [comment]
Hello,

 

I went back into the game. the draft of my map was there but no changes were saved.

I edited it again and when I clicked submit, this time everything went as planed and I could choose between executive change or vote procedure.

In the future when editing the district map. I will try to follow my steps more thoroughly, so I could give you a more detailed report.

For now I dont think there will  be anything to add.

 

Some side notes are maybe that every active user could start a vote for this changes.

Or a Minister can do it by executive order.

 

Cheers Shaane

 
 

Gesendet: Montag, 27. Dezember 2021 um 17:40 Uhr
Von: "tomesc99" ***@***.***>
An: "StrangeLoopGames/EcoIssues" ***@***.***>
Cc: "WickedFreaK" ***@***.***>, "Mention" ***@***.***>
Betreff: Re: [StrangeLoopGames/EcoIssues] USER ISSUE: Submiting District Map at the Zoning office (Issue #22870)


 

Hey @WickedFreaK ,

Thank you very much for your feedback. Did you try again after the crash? Were you able to reproduce the issue again?

If you are able to reproduce the issue again, could you provide the log files?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tomesc99** [comment]
Hello again, @WickedFreaK ,

Thank you for your answer. We are glad that the issue has no longer caused problems for you, but we are going to take it seriously.

Thank you very much for the feedback as this helps to improve the game.

**Comment by TheDu42** [comment]
FWIW i've had this issue several times in the past, across multiple versions.  seems like if you add to much area to new districts in one update something hangs up.  first time i ran into it was when drawing road districts across the whole map on a large world.  tried drawing the entire roadmap 3 times, and getting kicked with an error.  the workaround was to draw the districts in smaller chunks, only had the issue when adding too much area in one update

**Comment by IronyTroll** [comment]
I can also contribute that I have experienced this several times. Especially when modifying/adding a lot of area in the district draft. 

Say I add 1000 claims worth of district, hit save, then I get disconnected from the server and my changes aren't saved in the draft. 

In the short term, the only way we've got around this is by making smaller changes, say under 100 claims, then saving, then making 100 claim changes, then saving etc.
